src/knossos/model.clj

```diff
@@ -89,6 +89,61 @@
   ([]      (CASRegister. nil))
   ([value] (CASRegister. value)))
 
+(defrecord CausalRegister [value counter last-pos]
+  Model
```

Comment:
Are CausalRegisters usable with linearizability checking? And can the causal checker take regular Knossos registers? Or are these two distinct types of model?

Reply:
I didn't actually think of that while I was making this, but a linearizable checker would likely be able to use a causal register, so long as the test producing the history was shaped with the right values. The causal checker does little more than filter to ok ops and advance the history through the model, so I don't think it would be very useful with a regular register. Still should be compatible though!